air-quality-ui/src/services/measurement-data-service.ts

```diff
@@ -1,10 +1,52 @@
 import { DateTime } from 'luxon'
 
-import { MeasurementSummaryResponseDto } from './types'
+import { MeasurementSummaryResponseDto, MeasurementsResponseDto } from './types'
 import { LocationType } from '../models'
 
 const API_URL = import.meta.env.VITE_AIR_QUALITY_API_URL
 
+const queryMeasurements = (
+  urlString: string,
+  params: Record<string, string | string[]>,
+) => {
+  const url = new URL(`${API_URL}/air-pollutant${urlString}`)
+  Object.keys(params).forEach((key) => {
+    const value = params[key]
+    if (typeof value === 'string') {
+      url.searchParams.append(key, value)
+    } else {
+      value.forEach((v) => url.searchParams.append(key, v))
+    }
+  })
+  return fetch(url.toString(), {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  }).then((response) => response.json())
+}
```

Comment:
Is there value in pulling this out into a common file/function to be used across services? I think the forecast-data-service has some similar code.

Reply:
This makes sense yeah - I'll do it in the next one!